### PR TITLE
Correction doc for some Series.str methods

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_stringmethods_functions.py [hunk 1]
```diff
@@ -913,27 +913,25 @@ def _hpat_pandas_stringmethods_autogen(method_name):
     return _hpat_pandas_stringmethods_autogen_global_dict[global_dict_name]
 
 
-@overload_method(StringMethodsType, 'istitle')
-def hpat_pandas_stringmethods_istitle(self):
-    """
+hpat_pandas_series_str_is_docstring_tmpl = """
```
Comment (kozlov-alexey) [inline sdc/datatypes/hpat_pandas_stringmethods_functions.py]: I think better to name it "sdc_pandas_series_str_docstring_template"

File: sdc/datatypes/hpat_pandas_stringmethods_functions.py [hunk 2]
```diff
@@ -1227,80 +1046,44 @@ def hpat_pandas_stringmethods_islower_impl(self):
...
+        return pandas.Series(result, self._data._index, name=self._data._name)
 
-        Intel Scalable Dataframe Compiler Developer Guide
-        *************************************************
+    return hpat_pandas_stringmethods_isalnum_impl
 
-        Pandas Series method :meth:`pandas.core.strings.StringMethods.isalnum()` implementation.
 
-        Note: Unicode type of list elements are supported only. Numpy.NaN is not supported as elements.
+hpat_pandas_stringmethods_istitle.__doc__ = hpat_pandas_series_str_is_docstring_tmpl.format(**{
```
Comment (kozlov-alexey) [inline sdc/datatypes/hpat_pandas_stringmethods_functions.py]: I think it's better to do this in a loop, iterating over keys, values of a dict like:
stringmethods_funcs = {
    'istitle':  hpat_pandas_stringmethods_istitle,
     ...
}

File: sdc/datatypes/hpat_pandas_stringmethods_functions.py [hunk 3]
```diff
@@ -1069,9 +1069,18 @@ def hpat_pandas_stringmethods_isalnum_impl(self):
...
+    'isalpha': 'Check whether all characters in each string are alphabetic',
+    'islower': 'Check if all the characters in the text are alphanumeric',
+    'isalnum': 'Check if all the characters in the text are alphanumeric'
+}
+
 
 for name, method in stringmethods_funcs.items():
-    method.__doc__ = sdc_pandas_series_str_docstring_template.format(**{'method_name': name})
+    method.__doc__ = sdc_pandas_series_str_docstring_template.format(**{'method_name': name,
+                                                                        'caption': stringmethods_captions[name]})
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_stringmethods_functions.py]:
```
for method_name, caption in stringmethods_funcs.items():
   # ...
```

File: sdc/datatypes/hpat_pandas_stringmethods_functions.py [hunk 4]
```diff
@@ -1303,6 +1061,28 @@ def hpat_pandas_stringmethods_isalnum_impl(self):
     return hpat_pandas_stringmethods_isalnum_impl
 
 
+stringmethods_funcs = {
+    'istitle': hpat_pandas_stringmethods_istitle,
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_stringmethods_functions.py]: Let's merge the dicts to a single one, e.g.:
```
{
    'istitle': {
        'method': hpat_pandas_stringmethods_istitle,
        'caption': 'Check if each word start with an upper case letter'
    }
}
```

**Pull request merged**